File: be/src/main/java/sg/flow/entities/Bank.java
package sg.flow.entities;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@AllArgsConstructor
@Builder
@Table(name = "banks")
public class Bank {

    @Id
    private Integer id;

    @NotNull
    private String name;

    @NotNull
    private String bankCode;
}

File: be/src/main/kotlin/sg/flow/entities/Bank.kt
package sg.flow.entities

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.NotBlank
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

data class Bank(
    @field:Id
    val id: Int?,

    @field:NotBlank
    val name: String,

    @field:NotBlank
    val bankCode: String,

    val finverseId: String? = null,

    val countries: String = "SGP"
) 